### PR TITLE
Feature/160 Discussion block

### PR DESCRIPTION
**Pull request by Quipex**
Contains various bugfixes for the entire website, sorry that not in separate branches. It would take too much time for me to separate the commits by branches

Fixes up video naming, links, previews and length for the pre-created data

Closes:
#160 Discussions block
#341 If all lectures are viewed - the course is not displayed completed in the Student profile
#258 Video description on lectures is jammed
#305 Full watched lecture displayes less than 100%
#332 Wrap name of course to the next line

### CODE REVIEW

File: backend/core/src/main/resources/application.properties [hunk 1]
```diff
@@ -65,3 +65,10 @@ kafka.bootstrapAddress=127.0.0.1:9092
 kafka.topics.fileprocessor=process_file
 kafka.topics.notifications=notifications
 kafka.topics.emailsender=send_email
+
+#-------------------------
+# Notification demo
+#-------------------------
+test.notification.token=70ea66d1-8491-4b4a-b934-960abd7a84c7
```
Comment (AndersenGans) [inline backend/core/src/main/resources/application.properties:72]: required?

File: backend/core/src/test/resources/application.properties [hunk 2]
```diff
@@ -59,3 +59,10 @@ kafka.bootstrapAddress=127.0.0.1:9092
 kafka.topics.fileprocessor=process_file
 kafka.topics.notifications=notifications
 kafka.topics.emailsender=send_email
+
+#-------------------------
+# Notification demo
+#-------------------------
+test.notification.token=70ea66d1-8491-4b4a-b934-960abd7a84c7
```
Comment (AndersenGans) [inline backend/core/src/test/resources/application.properties:66]: required?

File: backend/push/env.ts [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
-        tokenSecret: getOsEnv('TOKEN_SECRET'),
-        port: getOsEnv('PUSH_APP'),
-        bootstrapAddress: getOsEnv('BOOTSTRAP_ADDRESS'),
-        consumerGroup: getOsEnv('CONSUMER_GROUP'),
-        notificationsTopic: getOsEnv('NOTIFICATIONS_TOPIC')
+        tokenSecret: 'mdzLa0RXxQX8mL9hFjctK9hCTDW4BQXHZXQZPPAKdD0LR6K0rB9rJrb6rM1otpWGXVABmxH0n4G1oFkp5Z03fGMXpvi3yeMq9jgxpnnNU6vP0j2UneEdm8Me1JK9qEQV',
+        port: '3003',
+        bootstrapAddress: '127.0.0.1:9092',
+        consumerGroup: 'notifiactions_group',
+        notificationsTopic: 'notifications'
```
Comment (AndersenGans) [inline backend/push/env.ts]: what was wrong with previous implementation?

File: backend/push/src/websockets/socket.service.ts [hunk 4]
```diff
@@ -11,21 +13,21 @@ export class SocketService {
...
         this.io.use((socket: any, next: any) => {
-            const token = socket.handshake.query.token;
-            const { valid, error } = isValid(token);
-            if (valid) {
-                return next();
-            }
-            console.log(`While connecting user with token: ${token} an error was caught: ${error}`);
-            return next(error);
+            const { token } = socket.handshake.query;
+            console.log(`token ${token}`);
```
Comment (AndersenGans) [inline backend/push/src/websockets/socket.service.ts]: console?

File: frontend/src/containers/WebSocketNotifications/index.tsx [hunk 5]
```diff
@@ -1,38 +1,46 @@
...
-      notify({ notification: message });
-    });
-    //  diconnect and delete from pool at logout
+    console.log('connection', connection);
+    if (connection) {
+      connection.on('notification', (message: INotification) => {
+        NotificationManager.info(message.text);
+        notify({ notification: message });
+      });
+      console.log('connected to notifications ws room');
```
Comment (AndersenGans) [inline frontend/src/containers/WebSocketNotifications/index.tsx]: console?

File: frontend/src/containers/WebSocketNotifications/index.tsx [hunk 6]
```diff
@@ -1,38 +1,46 @@
...
+    }
     return () => {
-      socket.close();
+      if (connection) {
+        connection.removeAllListeners('notification');
+        console.log('disconnected notifications ws room');
```
Comment (AndersenGans) [inline frontend/src/containers/WebSocketNotifications/index.tsx]: console?

File: frontend/src/containers/discussions/CourseDiscussion/index.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+  receiveMessage, websocketConnection: io
+}) => {
+  const handleSendMessage = useCallback(text => sendMessage({ text, resourceId: courseId }), [courseId]);
+
+  useEffect(() => {
+    io.on(`course:${courseId}`, (message: IMessage) => {
+      console.log('received message from ws', message);
+      receiveMessage(message);
+    });
+    console.log(`connected to ws course:${courseId}`);
```
Comment (AndersenGans) [inline frontend/src/containers/discussions/CourseDiscussion/index.tsx]: console?

File: frontend/src/containers/discussions/CourseDiscussion/index.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    return () => {
+      io.removeAllListeners(`course:${courseId}`);
+      console.log(`disconnect from ws room course:${courseId}`);
```
Comment (AndersenGans) [inline frontend/src/containers/discussions/CourseDiscussion/index.tsx]: console?

File: frontend/src/containers/discussions/LectureDiscussion/index.tsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+
+  useEffect(() => {
+    io.on(`lecture:${lectureId}`, (message: IMessage) => {
+      console.log('received message from ws', message);
+      receiveMessage(message);
+    });
+    console.log(`connected to ws lecture:${lectureId}`);
+    return () => {
+      io.removeAllListeners(`lecture:${lectureId}`);
+      console.log(`disconnect from ws room lecture:${lectureId}`);
```
Comment (AndersenGans) [inline frontend/src/containers/discussions/LectureDiscussion/index.tsx]: console?

File: backend/core/src/test/resources/application.properties (hunk 2 above)
Comment (Quipex) [reply backend/core/src/test/resources/application.properties:66]: Yes

File: backend/core/src/main/resources/application.properties (hunk 1 above)
Comment (Quipex) [reply backend/core/src/main/resources/application.properties:72]: Yes

File: backend/push/env.ts (hunk 3 above)
Comment (Quipex) [reply backend/push/env.ts]: Oops, my bad, thanks for noticing

File: backend/push/src/websockets/socket.service.ts (hunk 4 above)
Comment (Quipex) [reply backend/push/src/websockets/socket.service.ts]: removed

File: frontend/src/containers/WebSocketNotifications/index.tsx (hunk 5 above)
Comment (Quipex) [reply frontend/src/containers/WebSocketNotifications/index.tsx]: removed

File: frontend/src/containers/WebSocketNotifications/index.tsx (hunk 6 above)
Comment (Quipex) [reply frontend/src/containers/WebSocketNotifications/index.tsx]: removed

File: frontend/src/containers/discussions/CourseDiscussion/index.tsx (hunk 7 above)
Comment (Quipex) [reply frontend/src/containers/discussions/CourseDiscussion/index.tsx]: removed

File: frontend/src/containers/discussions/CourseDiscussion/index.tsx (hunk 8 above)
Comment (Quipex) [reply frontend/src/containers/discussions/CourseDiscussion/index.tsx]: removed

File: frontend/src/containers/discussions/LectureDiscussion/index.tsx (hunk 9 above)
Comment (Quipex) [reply frontend/src/containers/discussions/LectureDiscussion/index.tsx]: removed

**Review by AndersenGans — APPROVED** [review]

**Pull request merged**